### ISSUE TITLE
False positive 'Expected "else"' on `A if B != C else D` in f-string expression

### ISSUE THREAD

**Issue by gthb** [microsoft/pyright#526]
**Describe the bug**
If an `{expression}` inside an `f'...'` format string contains an `A if B else C` expression where the `B` subexpression contains the operator `!=`, then pyright gets confused and reports `Expected "else"` at the space before the `!=` operator.

**To Reproduce**

Put this in a file and run pyright on that file:

```python
print(f"{'a' if 'b' != 'c' else 'd'}")
print('a' if 'b' != 'c' else 'd')
print(f"{'b' != 'c'}")
print(f"{'a' if 'b' == 'c' else 'd'}")
print(f"{'a' if 2 < 3 else 'd'}")
print(f"{'a' if 2 > 3 else 'd'}")
print(f"{'a' if 2 & 3 else 'd'}")
print(f"{'a' if 2 | 3 else 'd'}")

```

Instead, Pyright reports:

```
  1:20 - error: Expected "else"
```

It reports no errors or warnings on the other lines, so the only case in which I can reproduce this the pretty specific case of an `!=` operator appearing within an `A if B else C` expression appearing within an f-string.

**Expected behavior**

0 errors, 0 warnings

**VS Code extension or command-line**

Discovered while using pyright from the command line, but I see the same in VS Code. Pyright version 1.1.22.

**Comment by erictraut** [comment]
Thanks for the bug report. PEP 498 (which defines f-strings) explicitly states that "!" characters are not allowed in f-string expressions, but it makes an exception for the "!=" sequence. Pyright contained code to handle this exception, but there was a bug in it. I just fixed the bug and added a test case for it.

This will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is included in 1.1.23, which I just published.
